tests-integration/json-schema/test/json-schema.test.js

```diff
@@ -0,0 +1,202 @@
+    })
+})
+
+describe('Test Suite for Advanced Types', () => {
+
+    test.skip('1. Test for Decimal type', () => {
+        const decimalSchema = {
+           "type": "object",
+            "properties": {
+                "decimal": {
+                    "$ref": "https://morphir.finos.org/test_model.schema.json#/$defs/AdvancedTypes.Score"
+                }
+            }
+        }
+        expect(validatorBasic(decimalSchema, {decimal: "78.9"})).toBe(true)
+    })
+
+    test.skip('2. Test for LocalDate type', () => {
+    })
+
+    test.skip('3. Test for LocalTime type', () => {
+    })
+
+    test.skip('4. Test for Month type', () => {
+    })
+})
+
+describe('Test Suite for Optional Types', () => {
+    test('Test for MayBe String', () => {
+        const optionalSchema = jsonObject["$defs"]["OptionalTypes.Assignment"]
+        expect(validator(optionalSchema, "Foo")).toBe(true)
+    })
+})
+
+describe('Test Suite for Collection Types', () => {
+    test('Test for List type', () => {
+        const listSchema = jsonObject["$defs"]["CollectionTypes.Department"]
+        expect(validator(listSchema, ["Foo", "Bam", "Baz"])).toBe(true)
+    })
+
+    test('Test for Set type', () => {
+        const setSchema = jsonObject["$defs"]["CollectionTypes.Queries"]
+        expect(validator(setSchema, ["Foo", "Bam"])).toBe(true)
+    })
+
+    test.skip('Test for Dict type', () => {
+    })
+})
+
+describe('Test Suite for Result Types', () => {
+    test.skip('Test for  type', () => {
+    })
+})
+
+describe('Test Suite for Composite Types - Records/Tuples', () => {
+    test.skip('Test for Tuple  type', () => {
+    })
+
+    test('Test for Record type', () => {
+        const recordSchema = jsonObject["$defs"]["RecordTypes.Address"]
+        const recordInstance = {
+            domainName : "Foo",
+            path : "Bar",
+            protocol : "http"
+        }
+        expect(validator(recordSchema, recordInstance)).toBe(true)
+    })
+
+    test.skip('Test for Tuple type', () => {
+    })
+})
+
+describe('Test Suite for Composite Types - Custom Types', () => {
+    test('Test for Enum Type', () => {
+        const enumSchema = jsonObject["$defs"]["CustomTypes.Currencies"]
+        expect(validator(enumSchema, "EUR")).toBe(true)
+    })
+
+    test('Test for Custom type 1',  () => {
+        const custom1Schema = jsonObject["$defs"]["CustomTypes.Person"]
+        expect(validator(custom1Schema, ["Child", "Bar", 11])).toBe(true)
+    })
+
+    test('Test for Custom type 2',  () => {
+        const custom2Schema = jsonObject["$defs"]["CustomTypes.Person"]
+        expect(validator(custom2Schema, ["Adult", "Bar"])).toBe(true)
+    })
+})
+
+const validator = (schema, instance) => {
+    const ajv = new ajv2020()
+    const validate = ajv.compile(schema)
+    return validate( instance);
+}
+
+//For validation of Basic Types References the main schema which is JsonObject
+const validatorBasic = (schema, instance) => {
```

Comment:
It would be better to call this `validateBasicType`

Reply:
Done!